esphome/components/max31865/max31865.cpp

```diff
@@ -0,0 +1,171 @@
+#include "max31865.h"
+
+#include "esphome/core/log.h"
+#include <cmath>
+
+// NOTES: Currently configured for 3-wire PT1000 with Vbias always on
+
+#define MAX31865_WRITE (0x80)
+#define MAX31865_CFG (0x00)
+#define MAX31865_RTD_H (0x01)
+#define MAX31865_RTD_L (0x02)
+#define MAX31865_FLT (0x07)
+
+#define MAX31865_CFG_6050 (0x01)  // Filter (0 - 60 Hz / 1 - 50Hz)
+#define MAX31865_CFG_FLTC (0x02)  // 1 - Clear Fault (Auto-Clear)
+#define MAX31865_CFG_WIRE (0x10)  // Wires (0 - 2 or 4 Wire / 1 - 3 Wire)
+#define MAX31865_CFG_MODE (0x40)  // 1 - Auto conversion
+#define MAX31865_CFG_BIAS (0x80)  // 1 - Vbias Enabled
+#define MAX31865_CFG_WIRE_EVEN(x) x ? 0 : MAX31865_CFG_WIRE
+
+#define MAX31865_RTD_L_FLT (0x01)  // Indicates Fault
+
+#define MAX31865_FLT_OVUV (0x04)
+#define MAX31865_FLT_RTNO (0x08)
+#define MAX31865_FLT_RFNO (0x10)
+#define MAX31865_FLT_RFNS (0x20)
+#define MAX31865_FLT_TSHL (0x40)
+#define MAX31865_FLT_TSHH (0x80)
+
+#define RTD_A 3.9083e-3
+#define RTD_B -5.775e-7
+
+namespace esphome {
+namespace max31865 {
+
+static const char *TAG = "max31865";
+
+float MAX31865Sensor::temperature(unsigned short adc, float rtd_nominal, float ref_resistor) {
+  // https://github.com/adafruit/Adafruit_MAX31865/blob/master/Adafruit_MAX31865.cpp
+  // http://www.analog.com/media/en/technical-documentation/application-notes/AN709_0.pdf
+
+  float z1, z2, z3, z4, r_t, temp;
+
+  r_t = adc;
+  r_t /= 32768;
+  r_t *= ref_resistor;
+
+  ESP_LOGD(TAG, "'%s': Resistance: %.3f", this->name_.c_str(), r_t);
```

Comment:
```suggestion
  ESP_LOGV(TAG, "'%s': Resistance: %.3f", this->name_.c_str(), r_t);
```

Not really d level (d is for anything a normal user might be interested in, yeah i know it's not ideal naming but it is what it is)